core/jobs/batch_jobs/exp_migration_jobs.py

```diff
@@ -340,191 +344,6 @@ def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
         )
 
 
-# TODO(#15927): This job needs to be kept in sync with MigrateExplorationJob and
-# later we will unify these jobs together.
-class AuditExplorationMigrationJob(base_jobs.JobBase):
```

Comment:
Now that we are deleting the audit job, how does one execute the migration job in "audit" mode via the UI?

Reply:
@kevintab95 The idea behind this change is that we won't need an audit job in the first place because any migration errors will cause the entire migration to be skipped (using the filter results transform). Hence the datastore will always have valid models.

Reply:
Thanks @JayVivarekar! I think we might need a bit of clarity on this. While it is great that we can have just one job that does the migration in an atomic manner i.e. all models are committed or none at all, I think that we might still need a way to run the beam job without committing to the datastore even if the migration is successful. We want to be able to test / deploy things on the server without having to irreversibly updating the models there. So we might want a "flag" or similar in the beam job that will allow it to be run in either migrate mode or audit mode. @vojtechjelinek has this been discussed previously and if not, should we consider having 1 beam job but have 2 modes of operation as suggested?

Reply:
@kevintab95 Hmm, can you list some usages of the pure audit job? It wasn't discussed before.

Reply:
Given that resetting the server is an expensive operation, we need the to freedom to run a job in audit mode so that we don't irreversibly update the models.

Reply:
Okay, I'm fine with this. I think the best approach would be to extract most of the parts of the job into a transform and then have two jobs but only call the transform which can be modified. We have a meeting with @JayVivarekar tomorrow so we can discuss this.

Reply:
> Okay, I'm fine with this. I think the best approach would be to extract most of the parts of the job into a transform and then have two jobs but only call the transform which can be modified. We have a meeting with @JayVivarekar tomorrow so we can discuss this.

Done.